File: smsls.pl
#!/usr/bin/perl -w
#$Id$
use strict;
use constant DEBUGGING => 0;
use constant HOME => "/home/cmcfaul/Documents/"; #these should change
use constant DATADIR => "/home/cmcfaul/data/"; #with the machine

my @cell_center = (20, 89, 155, 221, 285, 351, 418, 486); # one point for each cell
my @range = -10..10;
my $step = 3;
my $cell = 1;
my @pixels = ();
#my @pixels = 1..39; # for cell 01
#my @pixels = 202..240; # for cell 04
#my @pixels = 467..505; # for cell 08
foreach (@range) {
	 my $test = $cell_center[$cell - 1] + $_*$step;
	 push(@pixels, $test) unless ($test < 0 || $test > 511);
}
@pixels = @cell_center; #to use one point for each cell

print "Created by $Id$\n ";
my $files = 0;
my @offset = (0);
while(<>) {
	my @data = split /\t/, $_;
	my $num = @data;
	if ($num == 1) { }
	elsif ($data[0] =~ /Time/) {
		unless ($files) {
			print $data[0] . "\t";
			foreach (@pixels) {
				print $data[$_] . "\t";
			}
			print "\n";
		}	
		$files ++;
	}
	else {
		print $data[0] + $offset[$files-1] . "\t";
		foreach (@pixels) {
			print $data[$_] . "\t";
		}
	print "\n";
	$offset[$files] = $offset[$files-1] + $data[0];
	}
}

File: smsls.py
#!/usr/bin/python
#$Id$
from numpy import *

datadir = 'C:\\Users\\cmcfaul\\Documents\\Data\\'
datedir = raw_input('Enter the date of the data (in yyyy-mm-dd) ')
basename = raw_input('What is the basename of the file? ') #'c121031_stab_rev'
outFile = []
for i in arange(8):
	f = open(datadir + datedir + '\\' + basename + '_c0' + str(i+1) + '.dat', 'w')
	outFile.append(f)

#setup the array to hold the pixels we want to keep
step = 3
cell_center = array ([20, 89, 155, 221, 285, 351, 418, 486])
cell_range = step*arange(-5, 6, 1)
rangex,cellx = ix_(cell_range, cell_center)
pixels = rangex+cellx

tempfile = open(datadir + datedir + '\\' + basename + '_temperatures.txt')
tempfile.readline() #header size
tempfile.readline() #version
header = tempfile.readline() #sampling interval
interval = int(header.split(':')[1])
tempfile.readline() #sampling rate
tempfile.readline() #sample count
tempfile.readline() #device serial number
tempfile.readline() #column titles

datafile = open(datadir + datedir + '\\' + basename + '.txt')
datafile.readline() #sentinal line

header = datafile.readline() #column titles
column = header.split('\t')
for i in arange(8):
	outFile[i].write('Time (s), Temperature (C), ' , )
	for idx in pixels[:,i]:
		outFile[i].write( str(column[idx+1]) + ', ', )
	outFile[i].write('\n')
for line in iter(datafile):
	ls = line.split('\t')
	data = tempfile.readline()
	temp = data.split(',')
	for i in arange(8):
		outFile[i].write(str(ls[0]) + ', ' , )
		outFile[i].write(str(temp[i+2]) + ', ' , )
		for idx in pixels[:,i]:
			outFile[i].write(str(ls[idx+1]) + ', ' , )
		outFile[i].write('\n')
for i in arange(8):
	outFile[i].close()
	
tempfile.close()
datafile.close()
